### PR TITLE
ZkClient 超时时间太短导致在简易的zookeeper开发集群上连接超时错误

### PR DESCRIPTION
**Pull request by GinRyan**
延长5000到15000，以防止ZkClient 连接Zk开发集群上的时候连接超时导致无法启动服务器的问题。
该问题在cim连接Docker Compose中架设的zk集群时会发生。延长时长可解决该问题。

### CODE REVIEW

**Comment by GinRyan** [comment]
该问题不是ZooKeeper连接库的问题，是ZkClient 0.1的问题。
连接库我改成依赖ZooKeeper:3.3.6，并且在docker集群上搭建了一个同样为3.3.6版本的ZooKeeper集群，连接同样存在这个问题。而我使用ZooKeeper 3.3.3 库中的类直接连接可以顺利连接ZooKeeper集群，改用ZkClient 却会出现问题。说明超时问题出在ZkClient 这个封装库本身。于是延长了超时时间，发现延长超时时间后继续等待可以连接上ZooKeeper集群，说明ZkClient 在5000ms超时时间以内就判定了连接超时错误。

**Comment by GinRyan** [comment]
问题堆栈log：
Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'ZKit': Unsatisfied dependency expressed through field 'zkClient'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'buildZKClient' defined in class path resource [com/crossoverjie/cim/server/config/BeanConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.I0Itec.zkclient.ZkClient]: Factory method 'buildZKClient' threw exception; nested exception is org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 5000

详细问题堆栈：
2019-02-17 15:11:08.559  INFO 16772 --- [           main] org.apache.zookeeper.ZooKeeper           : Session: 0x100001108d30004 closed                                                       2019-02-17 15:11:08.560  INFO 16772 --- [ain-EventThread] org.apache.zookeeper.ClientCnxn          : EventThread shut down                                                                   2019-02-17 15:11:08.561  WARN 16772 --- [           main] ationConfigEmbeddedWebApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'ZKit': Unsatisfied dependency expressed through field 'zkClient'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'buildZKClient' defined in class path resource [com/crossoverjie/cim/server/config/BeanConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.I0Itec.zkclient.ZkClient]: Factory method 'buildZKClient' threw exception; nested exception is org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 5000
2019-02-17 15:11:12.991  INFO 16772 --- [           main] c.c.cim.server.server.CIMServer          : 鍏抽棴 cim server 鎴愬姛                                                                2019-02-17 15:11:12.994  INFO 16772 --- [
   main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2019-02-17 15:11:13.021  INFO 16772 --- [           main] utoConfigurationReportLoggingInitializer :
Error starting ApplicationContext. To display the auto-configuration report re-run your application with 'debug' enabled.                                                                    2019-02-17 15:11:13.031 ERROR 16772 --- [
   main] o.s.boot.SpringApplication               : Application startup failed
                                                                                                                                                                                             org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'ZKit': Unsatisfied dependency expressed through field 'zkClient'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'buildZKClient' defined in class path resource [com/crossoverjie/cim/server/config/BeanConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.I0Itec.zkclient.ZkClient]: Factory method 'buildZKClient' threw exception; nested exception is org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 5000
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:588) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]                                      at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:366) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
                                                                                                   at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1264) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:553) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:761) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:867) ~[spring-context-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543) ~[spring-context-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.6.RELEASE.jar!/:1.5.6.RELEASE]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.6.RELEASE.jar!/:1.5.6.RELEASE]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.6.RELEASE.jar!/:1.5.6.RELEASE]                                                     at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.6.RELEASE.jar!/:1.5.6.RELEASE]                                                                at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.6.RELEASE.jar!/:1.5.6.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.6.RELEASE.jar!/:1.5.6.RELEASE]
        at com.crossoverjie.cim.server.CIMServerApplication.main(CIMServerApplication.java:30) [classes!/:1.0.0-SNAPSHOT]                                                                            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_201]                                                                                                               at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_201]
                                                                                                   at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_201]
                                                  at java.lang.reflect.Method.invoke(Unknown Source) ~[na:1.8.0_201]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [cim-server-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [cim-server-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]                                                                          at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [cim-server-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51) [cim-server-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'buildZKClient' defined in class path resource [com/crossoverjie/cim/server/config/BeanConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.I0Itec.zkclient.ZkClient]: Factory method 'buildZKClient' threw exception; nested exception is org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 5000                         at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1173) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
                                                                                                                                                    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1067) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
                                                  at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:208) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1138) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1066) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:585) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        ... 27 common frames omitted                                                                                                                                                         Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.I0Itec.zkclient.ZkClient]: Factory method 'buildZKClient' threw exception; nested exception is org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 5000
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        ... 39 common frames omitted
Caused by: org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 5000
        at org.I0Itec.zkclient.ZkClient.connect(ZkClient.java:876) ~[zkclient-0.1.jar!/:na]
        at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:98) ~[zkclient-0.1.jar!/:na]
        at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:92) ~[zkclient-0.1.jar!/:na]
        at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:76) ~[zkclient-0.1.jar!/:na]
        at com.crossoverjie.cim.server.config.BeanConfig.buildZKClient(BeanConfig.java:28) ~[classes!/:1.0.0-SNAPSHOT]
        at com.crossoverjie.cim.server.config.BeanConfig$$EnhancerBySpringCGLIB$$d9f64307.CGLIB$buildZKClient$1(<generated>) ~[classes!/:1.0.0-SNAPSHOT]
        at com.crossoverjie.cim.server.config.BeanConfig$$EnhancerBySpringCGLIB$$d9f64307$$FastClassBySpringCGLIB$$55cc5dbb.invoke(<generated>) ~[classes!/:1.0.0-SNAPSHOT]
        at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228) ~[spring-core-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:358) ~[spring-context-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        at com.crossoverjie.cim.server.config.BeanConfig$$EnhancerBySpringCGLIB$$d9f64307.buildZKClient(<generated>) ~[classes!/:1.0.0-SNAPSHOT]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_201]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_201]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_201]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[na:1.8.0_201]
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162) ~[spring-beans-4.3.10.RELEASE.jar!/:4.3.10.RELEASE]
        ... 40 common frames omitted

**Comment by crossoverJie** [comment]
本地无法模拟；不过既然改了超时时间，还是建议将它写到配置文件里。

**Comment by GinRyan** [comment]
> 
> 
> 本地无法模拟；不过既然改了超时时间，还是建议将它写到配置文件里。

这种可能因为机器性能的原因导致的或者是存在一些未知条件的情况确实是很难模拟……
连我都不知道我的机器到底特殊在哪……
看起来还是写在配置文件里不影响源码比较好

File: cim-forward-route/src/main/java/com/crossoverjie/cim/route/config/BeanConfig.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
     @Bean
     public ZkClient buildZKClient(){
-        return new ZkClient(appConfiguration.getZkAddr(), 5000);
+        return new ZkClient(appConfiguration.getZkAddr(), 15000);
```
Comment (crossoverJie) [inline cim-forward-route/src/main/java/com/crossoverjie/cim/route/config/BeanConfig.java]: 请修改为读取配置文件，方便其情况再做调整。

Comment (GinRyan) [reply cim-forward-route/src/main/java/com/crossoverjie/cim/route/config/BeanConfig.java]: 晚上回家改下

**Comment by GinRyan** [comment]
ok了

**Pull request merged**